### ISSUE TITLE
E1029 - false positive with X.509 Certificate AWS IoT Core policy variables

### ISSUE THREAD

**Issue by xeres** [aws-cloudformation/cfn-lint#2026]
cfn-lint version: (cfn-lint 0.49.2)

I got a false positive on X.509 Certificate AWS IoT Core policy variables.

I think that `cfn-lint` may not support X.509 Certificate AWS IoT Core policy variables.

https://docs.aws.amazon.com/iot/latest/developerguide/cert-policy-variables.html

As noted above, X.509 certificate attributes can contain one or more values, but `resource_excludes` does not support specification by regex. This problem is a similar case in the past, `${iot:Connection.Thing.Attributes[attributeName]}` could not be supported (see also #765).

However, I think that just supporting the first value covers many cases.

If you are okay with this idea, I can contribute.

Do you have any comments?

**Output:**
```shell-session
❯ cfn-lint iot-policy.cfn.yml
E1019 Parameter iot:Certificate.Subject.CommonName for Fn::Sub not found at Resources/IoTPolicy/Properties/PolicyDocument/Statement/1/Resource/0/Fn::Sub
iot-policy.cfn.yml:18:13
```

**For reproduction:**
```yaml
AWSTemplateFormatVersion: 2010-09-09

Resources:
  IoTPolicy:
    Type: AWS::IoT::Policy
    Properties:
      PolicyDocument:
        Version: 2012-10-17
        Statement:
          - Effect: Allow
            Action:
              - iot:Connect
            Resource:
              - arn:aws:iot:us-east-1:123456789012:client/${iot:Connection.Thing.ThingName}
          - Effect: Allow
            Action:
              - iot:Publish
            Resource:
              - !Sub "arn:aws:iot:us-east-1:123456789012:topic/my/topic/${iot:Certificate.Subject.CommonName}"
```

**Comment by kddejong** [comment]
@xeres when I use this template to create a stack i get `Template error: instance of Fn::Sub references invalid resource attribute iot:Certificate.Subject.CommonName` which is similar to the error we provide.

You will want to make sure you use the literal string like this `!Sub "arn:aws:iot:us-east-1:123456789012:topic/my/topic/${!iot:Certificate.Subject.CommonName}"`

This will make the error remove but it will bring back W1020 which says the sub isn't needed.

**Comment by xeres** [comment]
Oh, I'm sorry.
The reproduction template is wrong.

https://docs.aws.amazon.com/iot/latest/developerguide/certificate-policy-examples.html
A similar as the third case, but I want to use with `!Sub` for `${AWS::Partition}`, `${AWS::Region}` and `${AWS::AccountId}`.

```yaml
AWSTemplateFormatVersion: 2010-09-09

Resources:
  IoTPolicy:
    Type: AWS::IoT::Policy
    Properties:
      PolicyDocument:
        Version: 2012-10-17
        Statement:
          - Effect: Allow
            Action:
              - iot:Connect
            Resource:
              - arn:aws:iot:us-east-1:123456789012:client/${iot:Connection.Thing.ThingName}
          - Effect: Allow
            Action:
              - iot:Publish
            Resource:
              - !Join
                - ""
                - - !Sub "arn:${AWS::Partition}:iot:${AWS::Region}:${AWS::AccountId}:topic/my/topic/"
                  - "${iot:Certificate.Subject.CommonName}"
```
